docs/release-info/migr-mvd-array.md

```diff
@@ -0,0 +1,135 @@
+| Ingestion | <ul><li>JSON arrays are ingested as Druid arrays</li><li>Managed through the query context parameter `arrayIngestMode` in SQL-based ingestion (supported options: `array`, `mvd`, `none`). Note that if you set this mode to `none`, Druid raises an exception if you try to store any type of array.</li></ul> | <ul><li>JSON arrays are ingested as multi-value dimensions</li><li>Managed using functions like [ARRAY_TO_MV](../querying/sql-functions.md#array_to_mv) in SQL-based ingestion</li></ul> |
+| Filtering and grouping | <ul><li>Filters and groupings match the entire array value</li><li>Can be used as GROUP BY keys, grouping based on the entire array value</li></ul> | <ul><li>Filters match any value within the array</li><li>Grouping generates a group for each individual value, similar to an implicit UNNEST</li></ul> |
+| Conversion | Convert an MVD to an array using [MV_TO_ARRAY](../querying/sql-functions.md#mv_to_array) | Convert an array to an MVD using [ARRAY_TO_MV](../querying/sql-functions.md#array_to_mv) |
+
+## Query differences between arrays and MVDs
+
+In SQL queries, Druid operates on arrays differently than MVDs.
+A value in an array column is treated as a single array entity (SQL ARRAY), whereas a value in an MVD column is treated as individual strings (SQL VARCHAR).
+
+For example, consider the same value, `['a', 'b', 'c']` ingested into an array column and an MVD column.
+In your query, you want to filter results by comparing some value with `['a', 'b', 'c']`.
+
+* For array columns, Druid only returns the row when an equality filter matches the entire array.  
+For example: `WHERE "array_column" = ARRAY['a', 'b', 'c']`.
+
+* For MVD columns, Druid returns the row when an equality filter matches any value of the MVD.  
+For example, any of the following filters returns the row for the query:  
+`WHERE "mvd_column" = 'a'`  
+`WHERE "mvd_column" = 'b'`  
+`WHERE "mvd_column" = 'c'`
+
+Note this difference between arrays and MVDs when you write queries that involve filtering or grouping.
+
+The following examples highlight a few analogous queries between arrays and MVDs.
+For more information and examples, see [Querying arrays](../querying/arrays.md#querying-arrays) and [Querying multi-value dimensions](../querying/multi-value-dimensions.md#querying-multi-value-dimensions).
+
+### Example: an element in array
+
+Filter rows that have a certain value in the array or MVD.
+
+#### Array
+
+```sql
+SELECT *
+FROM "array_example"
+WHERE ARRAY_CONTAINS(tags, 't3')
+```
+
+#### MVD
+
+```sql
+SELECT *
+FROM "mvd_example"
+WHERE tags = 't3'
+```
+
+### Example: overlap of two arrays
+
+Filter rows for which the array or MVD overlaps a reference array.
+
+#### Array
+
+```sql
+SELECT *
+FROM "array_example"
+WHERE ARRAY_OVERLAP(tags, ARRAY['t1', 't7'])
+```
+
+#### MVD
+
+```sql
+SELECT *
+FROM "mvd_example"
+WHERE MV_OVERLAP(tags, ARRAY['t1', 't7'])
+```
```

Comment:
@2bethere you mentioned adding an example for ARRAY_OVERLAP but these examples aren't so different between MVDs and arrays. Did you have another use case in mind?

Reply:
The example I was thinking is:
If you do WHERE mvd in ("t1", "t2") today, how do you do that with ARRAY_CONTAINS/ARRAY_OVERLAP
ANother case is if you do WHERE mvd  ="t1" AND mvd =  "t2" today, how do you do that with ARRAY_CONTAINS/ARRAY_OVERLAP

So not MV_OVERLAP -> ARRAY_OVERLAP. But converting WHERE conditions into ARRAY. Because I don't think WHERE array = "t1" AND array = "t2" works.

Reply:
might be useful to include array equality (e.g. `WHERE tags = ARRAY['t1', 't2', 't3']` is i think equivalent to `WHERE MV_TO_ARRAY(tags) = ARRAY['t1', 't2', 't3']`)

and array grouping
(`SELECT label, tags FROM "arrayExample" GROUP BY 1,2` and `SELECT label, MV_TO_ARRAY(tags) FROM "mvd_example" GROUP BY 1, 2`)

Reply:
@clintropolis if MVD equality checks individual strings, would this query always return zero results?
```
SELECT *
FROM "mvd_example"
WHERE tags = 't1' AND tags = 't2'
```
since a single string can't be two different values

Reply:
@2bethere I updated the existing example to use `WHERE tags = 't1' OR tags = 't2'`

Reply:
>@clintropolis if MVD equality checks individual strings, would this query always return zero results?

This one is tricky, in SQL yes, because SQL considers this a contradiction so "simplifies" it to always false so nothing matches. However, in native json queries you can write that filter, which provides a way to check that the mvd row contains all of the elements, similar to using `MV_CONTAINS` i suppose.